pkg/apply/apply.go

```diff
@@ -125,14 +125,14 @@ func ApplyObject(ctx context.Context, client cnoclient.Client, obj Object, subco
 		log.Printf("could not encode %s for apply", objDesc)
 		return fmt.Errorf("could not encode for patching: %w", err)
 	}
-	us, err := clusterClient.Dynamic().Resource(rm.Resource).Namespace(namespace).Patch(ctx, name, types.ApplyPatchType, data, patchOptions, subresources...)
-	if err != nil {
-		return fmt.Errorf("failed to apply / update %s: %w", objDesc, err)
-	}
-
 	// consider removing in OCP 4.18 when we know field manager 'cluster-network-operator' no longer possibly
 	// exists in any object from all upgrade paths
-	if isDepFieldManagerCleanupNeeded(subcontroller) {
+	// Retrieve the current state of the resource
```

Comment:
nit: ideally id put in a comment why we needed to do this and what OCP version it can be removed.

Reply:
isn't that what's up above in lines 128-130? although correct me if I'm wrong, but I think this can be removed in 4.16 right? 4.15 will have these changes from GA onward and any cluster that eventually upgrades to 4.15 would have this resolved, so 4.16 and above wouldn't need it.

But, does it hurt? I suppose in the future another field manager name change could occur?